xy/rdp_simplify.go

```diff
@@ -0,0 +1,140 @@
+package xy
+
+import (
+	"math"
+)
+
+const stride = 2
+
+// Adopted from https://github.com/paulmach/orb/blob/master/simplify/douglas_peucker.go
+// Original licence:
+//
+// The MIT License (MIT)
+// Copyright (c) 2017 Paul Mach Permission is hereby granted, free of charge, to
+// any person obtaining a copy of this software and associated documentation
+// files (the "Software"), to deal in the Software without restriction,
+// including without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to permit
+// persons to whom the Software is furnished to do so, subject to the following
+// conditions: The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software. THE SOFTWARE
+// IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+// COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+// WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR
+// IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+// SimplifyIndexes uses the Douglas-Peucker simplify method to simplify a 2D
+// flatCoords.
+//
+// Threshold is the distance between a point and the selected start
+// and end line segment. It returns the indexes of the points.
+//
+// Note that the indexes are based on points, So acesss to x, y pair should be:
+//  x:=flatCoords[i*stride]
+//  y:=flatCoords[i*stride+1]
+func SimplifyIndexes(flatCoords []float64, threshold float64) []int {
```

Comment:
This function should be renamed to `SimplifyFlatCoords` and take `stride` as a parameter.

Reply:
`func SimplifyIndexes(stride int, flatCoords []float64, threshold float64) []int`
or 
`func SimplifyIndexes(layout geom.Layout, flatCoords []float64, threshold float64) []int`

Reply:
See https://github.com/twpayne/go-geom/blob/master/flat.go#L351 for example.